mlrun/projects/project.py

```diff
@@ -3446,6 +3448,102 @@ def remove_custom_packager(self, packager: str):
         """
         self.spec.remove_custom_packager(packager=packager)
 
+    def create_api_gateway(
+        self,
+        name: str,
+        functions: Union[
+            list[
+                Union[
+                    RemoteRuntime,
+                    ServingRuntime,
+                ]
+            ],
+            Union[RemoteRuntime, ServingRuntime],
+        ] = [],
+        host: Optional[str] = None,
+        path: Optional[str] = "",
+        description: Optional[str] = "",
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        canary: Optional[list[int]] = None,
+    ) -> Union[mlrun.runtimes.api_gateway.APIGateway, None]:
+        """
+        Creates a Nuclio API Gateway. Nuclio docs here: https://docs.nuclio.io/en/latest/reference/api-gateway/http.html
+
+        :param name (str): API Gateway name.
+        :param functions (Union[list[Union[RemoteRuntime, ServingRuntime]], Union[RemoteRuntime, ServingRuntime]]):
+            List of Nuclio functions or Nuclio function.
+
+        :param host (str): API Gateway host (optional).
+        :param path (str): API Gateway path (optional).
```

Comment:
I think these 2 parameters should be better explained, and what values they get by default if not specified.
Also, for host, can the user specify any value?

Reply:
@gilad-shaham That's a very good question. I've checked in the nuclio UI, and we do not allow specifying it manually. However, technically, it's possible to set it manually when making an API request. I'll leave only the 'path' to be specified (it's possible to specify the 'path' in the UI), so we will be aligned with how it works on the UI side.

Reply:
@rokatyy @gilad-shaham  in Nuclio, the hostname that you see in the API gateway UI comes from the following template available in the frontend spec:
```
"defaultHTTPIngressHostTemplate": "{{ .ResourceName }}-{{ .ProjectName }}.{{ .Namespace }}.app.dev65.lab.iguazeng.com"
```

This template is configurable on the Nuclio service level, and Mlrun can tell Nuclio to use this template for ingresses by passing `"@nuclio.fromDefault"` (see the code section [here](https://github.com/nuclio/nuclio/blob/development/pkg/platform/kube/platform.go#L1982-L1983)).
However I am not sure it can be done the same with API gateways. If not - perhaps an improvement is required on Nuclio's side to allow it as well.

Reply:
@TomerShor as an option, we can allow passing empty host (it's not permitted now) and generate it on nuclio side. So mlrun won't take responsibility on this

Reply:
@rokatyy yes that's exactly what I meant by: "an improvement is required on Nuclio's side to allow it as well."
Can be left as a TODO for now